x/token/genesis.go

```diff
@@ -5,19 +5,33 @@ import (
 )
 
 type GenesisState struct {
+	Balances map[string]sdk.Int `json:"balances"`
 }
 
 func ValidateGenesis(data GenesisState) error {
+	for account := range data.Balances {
```

Comment:
I think we should check tokens value to be positive

Reply:
Theoretically it might be negative. E.g. the user created a post and someone downvoted it, the balance would be negative in this case.